src/locations/dMCDA.jl

```diff
@@ -0,0 +1,574 @@
+        pref_locations::NamedTuple, rankingsin::NamedTuple
+    )::Tuple
+
+# Arguments
+- `criteria_store` : contains all criteria for all intervention types.
+- `interventions` : has intervention type names as keys and specifies aggregation function for criteria_store for each intervention.
+- `params` : parameters for particular scenario run of the model.
+- `pref_locations` : previous time step's selection of locations for each intervention.
+- `thresholds` : specifies risk filter thresholds for criteria_store.
+- `n_iv_locs` : specifies number of locations to apply intervention at.
+- `distances` : n_locations*n_locations, specifies Haversine distance between each location in the set.
+- `minimum_distance` : specifies minimum allowed distance between selected locations.
+- `use_dist` : specifies whether distance sorting should be used for each intervention.
+- `int_logs` : specifies whether location selection should be performed for each intervention.
+- `rankingsin` : pre-allocated store for location rankings for each intervention.
+
+# Returns
+Tuple :
+    - `pref_locations` : n_iv_locs highest ranked locations for each intervention.
+    - `rankings` : n_locations ⋅ 3 matrix holding [location_id, seeding_rank, shading_rank],
+        Values of 0 indicate locations that were not considered
+"""
+function guided_location_selection(criteria_store::NamedDimsArray, interventions::NamedTuple,
+    params::NamedDimsArray, thresholds::NamedTuple, n_iv_locs::Int64,
+    distances::Matrix, minimum_distance::Float64, use_dist::NamedTuple, iv_logs::NamedDimsArray,
+    pref_locations::NamedTuple, rankingsin::NamedTuple
+)::Tuple
+    # location_id, seeding rank, shading rank
+    mcda_func = mcda_methods[Int64(params("guided"))]
+
+    # filter criteria prior to decision making
+    criteria_store = filter_decision_matrix(criteria_store, thresholds)
+
+    if isempty(criteria_store)
+        # if all rows filtered in risk filtering, abort
+        return pref_locations, rankingsin
+    end
+
+    for int_key in keys(interventions)
+
+        # if using intervention, create specific decision matrix
+        if iv_logs(int_key)
+
+            # get criteria matrix aggregation for particular intervention
+            criteria_store_temp = interventions[int_key](criteria_store)
+
+            # cap to number of locations left after risk filtration
+            n_iv_locs = min(n_iv_locs, length(criteria_store_temp.locations))
+            location_ids::Array{Int64} = criteria_store_temp.locations
+
+            rankings = rankingsin[int_key]
+
+            criteria_store_temp = criteria_store_temp[in.(criteria_store_temp.locations, [location_ids]), :]
+            n_locations_all::Int64 = length(location_ids)
+
+            if n_locations_all != 0
+                # create intervention matrix
+                S, ws = create_intervention_matrix(criteria_store_temp, params, String(int_key))
```

Comment:
I don't think this is working as intended.

`S` is always empty because `Domain.mcda_criteria` holds identifiers that do not contain the intervention type

e.g., `iv__wave_stress` instead of `iv__seed__wave_stress`

https://github.com/open-AIMS/ADRIA.jl/blob/e3ac4720319cab17c0f1f1b3484ab69762d763dc/src/Domain.jl#L99-L101

Reply:
Hi Takuya, the intervention type is not used to find the identfiers, only the name of the intervention.

Reply:
The name of the intervention does not contain the intervention type, so `S` is always empty.

> the intervention type is not used to find the identfiers, only the name of the intervention.

`params.factors` holds `iv__wave_stress` so in the below (inside `create_intervention_matrix()`):

```julia
iv_params = params.factors[occursin.("iv__", params.factors).&occursin.(iv_type, params.factors)]
```

`occursin.(iv_type, params.factors)` would always evaluate to `false`.

Reply:
The intervention type is used in the weights though, params.factors contains the weights. (see in Criteria, all the weights now contain the intervention types in the name)

Reply:
I did this because the weight is not unique to the criteria type, you need to be able to identify the intervention also

Reply:
`iv_key` is set to `:seed`

The function is called like so:

`create_intervention_matrix(criteria_store_temp, params, String(iv_key))`

The filter is used like so:

`iv_params = params.factors[occursin.("iv__", params.factors).&occursin.(iv_type, params.factors)]`

A given `params.factors` is `iv__wave_stress`. The above evaluates to false because "seed" does not occur in the identifier, leading to an empty `S` matrix.

I found this as tried to run a few scenarios and stepped through. Are you seeing something different?

Reply:
That is not how it was when I ran it, have you changed much since I opened the PR? I'll have a look

Reply:
You can go back to the original state the code was in when you opened the PR. But I'd rather you be working on something else for now.

Reply:
I ran it and it works fine in picking out the correct criteria, so not sure what has changed... but I'll leave this PR for now

Reply:
That's weird cos it's one of the first issues I found before any changes were made... but yes, work it out later.

Reply:
weird :(